client/my-sites/themes/themes-selection.jsx

```diff
@@ -261,14 +301,27 @@ class ThemesSelection extends Component {
 			shouldFetchWpOrgThemes,
 			wpOrgQuery,
 			wpOrgThemes,
+			isLoggedIn,
 		} = this.props;
 
 		const interlacedThemes = interlaceThemes( themes, wpOrgThemes, query.search, isLastPage );
 
+		const isDiscoveryEnabled =
+			( config.isEnabled( 'themes/discovery-lits' ) && isLoggedIn ) ||
+			( config.isEnabled( 'themes/discovery-lots' ) && ! isLoggedIn );
+
 		return (
 			<div className="themes__selection">
 				<QueryThemes query={ query } siteId={ source } />
 				{ shouldFetchWpOrgThemes && <QueryThemes query={ wpOrgQuery } siteId="wporg" /> }
+
+				{ isDiscoveryEnabled && (
+					<>
+						<PremiumThemesCollection>{ this.getCarouselBlock }</PremiumThemesCollection>
+						<PartnerThemesCollection>{ this.getCarouselBlock }</PartnerThemesCollection>
```

Comment:
I think this is super confusing and overcomplicated.

Throwing an idea into the mix: collection-specific **information** instead of collection-specific **components**.

Hopefully the example could clarify what I'm thinking:

_(Note: this is all untested!)_
```jsx
// theme-collection/constants.jsx

export const THEME_COLLECTIONS = {
  premium: {
    query: { /* ... */ },
    heading: translate( 'Premium Themes' ),
    subheading: <p>{ translate( 'Lorem ipsum dolor sit amet' ) }</p>,
  }
}
```

```jsx
// themes-selection.jsx

getThemeCollection = ( slug ) => {
  const {
    query,
    heading,
    subheading,
  } = THEME_COLLECTIONS[ slug ];
  
  const { themes } = useThemeCollection( query );

  return (
    <>
      <QueryThemes query={ query } siteId="wpcom" />
      <ThemeCollection
        slug={ slug }
        heading={ heading }
        subheading={ subheading }
      >
        { themes && themes.map( ( theme, index ) => (
          <ThemeCollectionItem key={ theme.id }>
            <ThemeBlock /* all the props */ />
          </ThemeCollectionItem>
        ) ) }
      </ThemeCollection>
    </>
  );
}

render() {
  return(
    <>
      { isDiscoveryEnabled &&
        getThemeCollection( 'premium' )
      }
    </>
  );
}
```

Off the top of my head, this has 2 big advantages:
- Simplifies the hell out of everything, reducing the prop drilling.
- Does not couple any theme fetching logic with collections.
  - By and large, we're going to show this in 1 or 2 places (TS and DP). I haven't checked, but I presume that DP has a completely different theme fetching logic (IIRC it uses a different endpoint even).
  - If we create a collection-specific component using `QueryThemes`, we are fundamentally limiting it to the TS, which would remove its usefulness altogether (what's the point of such a specific component if we are only going to use it once?).

Reply:
> THEME_COLLECTIONS.premium.query

I have some mixed feelings about keeping logic code in a constants file, separated from the actual code. I'm fine with keeping the texts (e.g. heading), but not sure about the query.

It's interesting. However, there's one problem with this approach. The code would add up and will increase even more the complexity of the ThemeSelection.

An alternative is to have a new component called `ThemeCollectionsLayout` that would:
- define the collections (query, headings, etc).
- construct the layout (collections, upgrade banners)

Reply:
> in a constants file

You could just call the file "collection-definitions.ts" if that helps

Reply:
> I have some mixed feelings about keeping logic code in a constants file

That really isn't logic though. It's the options argument for the themes endpoint. It would be the same as adding, like, `themesPerPage: 10`.